android/src/main/kotlin/com/signify/hue/flutterreactiveble/channelhandlers/ScanDevicesHandler.kt

```diff
@@ -15,7 +15,10 @@ class ScanDevicesHandler(private val bleClient: com.signify.hue.flutterreactiveb
     private var scanDevicesSink: EventChannel.EventSink? = null
     private lateinit var scanForDevicesDisposable: Disposable
     private val converter = ProtobufMessageConverter()
-    private var scanParameters: ScanParameters? = null
+
+    companion object {
+        private var scanParameters: ScanParameters? = null
+    }
```

Comment:
This is more localized, thus better, indeed. Still global state, unfortunately, but if you're saying that's really needed, so be it.

Reply:
yes unfortunately we need it there is no other way (at the moment) how to solve the issue